src/webgpu/api/operation/command_buffer/render/culling_tests.spec.ts

```diff
@@ -0,0 +1,160 @@
+    isCCWTriangleCulled: boolean,
+    isCWTriangleCulled: boolean,
+    format: GPUTextureFormat
+  ): void {
+    const size = 4;
+    const texture = this.create2DTextureForTest(format, size);
+    const encoder = this.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: texture.createView(),
+          loadValue: { r: 0.0, g: 0.0, b: 1.0, a: 1.0 },
+          storeOp: 'store',
+        },
+      ],
+    });
+    pass.setPipeline(this.createPipelineForTest(frontFace, cullMode, format));
+    pass.draw(6, 1, 0, 0);
+    pass.endPass();
+
+    const topLeftBuffer = this.device.createBuffer({
+      size,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+    });
+    encoder.copyTextureToBuffer(
+      { texture, mipLevel: 0, origin: { x: 0, y: 0, z: 0 } },
+      { buffer: topLeftBuffer, bytesPerRow: 256 },
+      { width: 1, height: 1, depth: 1 }
+    );
+
+    const bottomRightBuffer = this.device.createBuffer({
+      size,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+    });
+    encoder.copyTextureToBuffer(
+      { texture, mipLevel: 0, origin: { x: size - 1, y: size - 1, z: 0 } },
+      { buffer: bottomRightBuffer, bytesPerRow: 256 },
+      { width: 1, height: 1, depth: 1 }
+    );
+
+    this.device.defaultQueue.submit([encoder.finish()]);
+
+    const kCCWTriangleTopLeftColor = isCCWTriangleCulled
+      ? new Uint8Array([0x00, 0x00, 0xff, 0xff])
+      : new Uint8Array([0x00, 0x00, 0x00, 0xff]);
+    this.expectContents(topLeftBuffer, kCCWTriangleTopLeftColor);
+
+    const kCWTriangleBottomRightColor = isCWTriangleCulled
+      ? new Uint8Array([0x00, 0x00, 0xff, 0xff])
+      : new Uint8Array([0x03, 0x03, 0x00, 0xff]);
+    this.expectContents(bottomRightBuffer, kCWTriangleBottomRightColor);
+  }
+}
+
+export const g = makeTestGroup(CullingTest);
+
+g.test('CullNoneWhenCCWIsFrontFace')
+  .params([{ format: 'rgba8unorm' as GPUTextureFormat }])
+  .fn(t => {
+    t.run('ccw', 'none', false, false, t.params.format);
+  });
+
+g.test('CullFrontFaceWhenCCWIsFrontFace')
+  .params([{ format: 'rgba8unorm' as GPUTextureFormat }])
+  .fn(t => {
+    t.run('ccw', 'front', true, false, t.params.format);
+  });
+
+g.test('CullBackFaceWhenCCWIsFrontFace')
+  .params([{ format: 'rgba8unorm' as GPUTextureFormat }])
+  .fn(t => {
+    t.run('ccw', 'back', false, true, t.params.format);
+  });
+
+g.test('CullNoneWhenCWIsFrontFace')
+  .params([{ format: 'rgba8unorm' as GPUTextureFormat }])
+  .fn(t => {
+    t.run('cw', 'none', false, false, t.params.format);
+  });
+
+g.test('CullFrontFaceWhenCWIsFrontFace')
+  .params([{ format: 'rgba8unorm' as GPUTextureFormat }])
+  .fn(t => {
+    t.run('cw', 'front', false, true, t.params.format);
+  });
+
+g.test('CullBackFaceWhenCWIsFrontFace')
+  .params([{ format: 'rgba8unorm' as GPUTextureFormat }])
+  .fn(t => {
+    t.run('cw', 'back', true, false, t.params.format);
+  });
```

Comment:
We can parameterize over these different cases by doing something like this (totally untested):

```typescript
// use this in t.run() to compute the expectations:
function faceIsCulled(face: 'cw' | 'ccw', frontFace: GPUFrontFace, cullMode: GPUCullMode): boolean {
  return cullMode !== 'none' && ((frontFace === face) === (cullMode === 'front'));
}

g.test('cull_state')
  .params(
    params()
      .combine(poptions('frontFace', ['ccw', 'cw']))
      .combine(poptions('cullMode', ['none', 'front', 'back']))
  )
  .fn(t => {
    t.run(t.params.frontFace, t.params.cullMode);
  });
```

alternatively:

```typescript
g.test('cull_state')
  .params([
    { cullMode: 'none', frontFace: 'cw', _expectCCWCulled: false, _expectCWCulled: false },
    // ...
  ] as const)
  .fn(t => {
    t.run(t.params.frontFace, t.params.cullMode, t.params._expectCCWCulled, t.params._expectCWCulled);
  });
```

Note with either change, there would now be only one usage of t.run(), so if you don't end up with more, then you can just move its contents inline here.